### PR TITLE
Clean atom event listener

### PR DESCRIPTION
**Pull request by talekhinezh**
Remove API use of RE atom event listener, the connection between the two was too "strong". API should be accessing data through RE state store layer as the RE doesn't know when certain states are being virtualized.

### CODE REVIEW

File: radixdlt/src/main/java/org/radix/api/services/AtomsService.java [hunk 1]
```diff
@@ -155,6 +121,42 @@ public AtomsService(LedgerEntryStore store, SubmissionControl submissionControl,
 		});
 	}
 
+	public void start() {
+		io.reactivex.rxjava3.disposables.Disposable lastStoredAtomDisposable = engineStore.lastStoredAtom()
+			.observeOn(Schedulers.io())
```
Comment (flotothemoon) [inline radixdlt/src/main/java/org/radix/api/services/AtomsService.java]: What is this line for?

**Review by flotothemoon — COMMENTED** [review]
The implementation logic seems fine to me. I'm not quite clear yet how this will help improve the architecture - is this step just to decouple the RE more from core?

File: radixdlt/src/main/java/org/radix/api/services/AtomsService.java (hunk 1 above)
Comment (talekhinezh) [reply radixdlt/src/main/java/org/radix/api/services/AtomsService.java]: Setting the thread on which observations will occur on

File: radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java [hunk 2]
```diff
@@ -198,6 +202,8 @@ public final void start(RuntimeProperties properties) {
 	}
 
 	public final void stop() {
+    	this.atomsService.stop();
```
Comment (clstrfsck) [inline radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java]: Whitespace issues

File: radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java [hunk 3]
```diff
@@ -150,6 +152,8 @@ public RadixHttpServer(
     }
 
 	public final void start(RuntimeProperties properties) {
+    	this.atomsService.start();
```
Comment (clstrfsck) [inline radixdlt/src/main/java/org/radix/api/http/RadixHttpServer.java]: Whitespace issues

File: radixdlt/src/main/java/com/radixdlt/middleware2/store/CommittedAtomsStore.java [hunk 4]
```diff
@@ -76,11 +89,29 @@ public void getAtomContaining(Particle particle, boolean isInput, Consumer<Ledge
...
+
+        lastStoredAtom.onNext(storedEvent);
+    }
+
+    /**
+     * Retrieve a stream of the latest stored atoms
+     * @return hot observable of last stored atoms
+     */
+    public Observable<AtomStoredEvent> lastStoredAtom() {
+        return lastStoredAtom;
```
Comment (clstrfsck) [inline radixdlt/src/main/java/com/radixdlt/middleware2/store/CommittedAtomsStore.java:114]: Yeah, this is moar gooder.

File: radixdlt/src/main/java/org/radix/api/services/AtomsService.java [hunk 5]
```diff
@@ -155,6 +121,42 @@ public AtomsService(LedgerEntryStore store, SubmissionControl submissionControl,
 		});
 	}
 
+	public void start() {
+		io.reactivex.rxjava3.disposables.Disposable lastStoredAtomDisposable = engineStore.lastStoredAtom()
```
Comment (clstrfsck) [inline radixdlt/src/main/java/org/radix/api/services/AtomsService.java]: Less use of `Events` is nice.

Hopefully we can sort out the name clash on `Disposable` shortly.

File: radixdlt/src/main/java/org/radix/api/services/AtomsService.java [hunk 6]
```diff
@@ -155,6 +121,42 @@ public AtomsService(LedgerEntryStore store, SubmissionControl submissionControl,
...
+			.observeOn(Schedulers.io())
+			.subscribe(storedEvent -> {
+				LedgerAtom atom = storedEvent.getAtom();
```
Comment (clstrfsck) [inline radixdlt/src/main/java/org/radix/api/services/AtomsService.java]: Would prefer to see all this logic in a separate method.

**Review by clstrfsck — APPROVED** [review]
Looks good.  A couple of minor whitespace issues, and some preferences around inlining long lambdas could be considered.

File: radixdlt/src/main/java/com/radixdlt/middleware2/store/CommittedAtomsStore.java (hunk 4 above)
Comment (clstrfsck) [reply radixdlt/src/main/java/com/radixdlt/middleware2/store/CommittedAtomsStore.java:114]: (in reference to previous removal of AtomListener)

**Pull request merged**